lib/fishjam/local/room_service.ex

```diff
@@ -0,0 +1,235 @@
+defmodule Fishjam.Local.RoomService do
+  @moduledoc """
+  Module responsible for managing rooms on this Fishjam instance.
+  """
+
+  @behaviour Fishjam.RoomService
+
+  use GenServer
+
+  require Logger
+
+  alias Fishjam.Event
+  alias Fishjam.Local.Room
+  alias Fishjam.Room.ID
+  alias Fishjam.WebhookNotifier
+
+  @metric_interval_in_seconds Application.compile_env!(:fishjam, :room_metrics_scrape_interval)
+  @metric_interval_in_milliseconds @metric_interval_in_seconds * 1_000
+
+  ## NON-ROUTABLE FUNCTIONS
+
+  @spec start_link(term()) :: GenServer.on_start()
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  @spec get_resource_usage() :: %{
+          node: Node.t(),
+          forwarded_tracks_number: integer(),
+          rooms_number: integer()
+        }
+  def get_resource_usage() do
+    room_ids = get_rooms_ids()
+
+    room_ids
+    |> Enum.map(fn room_id ->
+      Task.Supervisor.async_nolink(Fishjam.TaskSupervisor, fn ->
+        Room.get_num_forwarded_tracks(room_id)
+      end)
+    end)
+    |> Task.yield_many()
+    |> Enum.map(fn {task, res} ->
+      res || Task.shutdown(task, :brutal_kill)
+    end)
+    |> Enum.filter(fn
+      {:ok, _res} -> true
+      _other -> false
+    end)
+    |> Enum.map(fn {:ok, res} -> res end)
+    |> then(
+      &%{node: Node.self(), forwarded_tracks_number: Enum.sum(&1), rooms_number: Enum.count(&1)}
+    )
+  end
+
+  ## CLUSTER-ROUTABLE FUNCTIONS
+
+  @impl Fishjam.RoomService
+  def find_room(room_id) do
+    case Registry.lookup(Fishjam.RoomRegistry, room_id) do
+      [{room_pid, _value}] ->
+        {:ok, room_pid}
+
+      _not_found ->
+        {:error, :room_not_found}
+    end
+  end
+
+  @impl Fishjam.RoomService
+  def get_room(room_id) do
+    room = Room.get_state(room_id)
+
+    if is_nil(room) do
+      {:error, :room_not_found}
+    else
+      {:ok, room}
+    end
+  end
+
+  @impl Fishjam.RoomService
+  def list_rooms() do
+    get_rooms_ids()
+    |> Enum.map(&Room.get_state(&1))
+    |> Enum.reject(&(&1 == nil))
+  end
+
+  @impl Fishjam.RoomService
+  def create_room(config) do
+    GenServer.call(__MODULE__, {:create_room, config})
+  end
+
+  @impl Fishjam.RoomService
+  def delete_room(room_id) do
+    GenServer.call(__MODULE__, {:delete_room, room_id})
+  end
+
+  ## GENSERVER CALLBACKS
+
+  @impl true
+  def init(_opts) do
+    {:ok, %{rooms: %{}}, {:continue, nil}}
+  end
+
+  @impl true
+  def handle_continue(_continue_arg, state) do
+    Process.send_after(self(), :rooms_metrics, @metric_interval_in_milliseconds)
+    :ok = Phoenix.PubSub.subscribe(Fishjam.PubSub, "fishjams")
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_call({:create_room, config}, _from, state) do
+    Logger.debug("Creating a new room")
+
+    # Re-generate the room ID to ensure it has the correct node name part
+    {:ok, room_id} = ID.generate(config.room_id)
```

Comment:
Oh shiet that was a huge bug lol. I feel like instead of having it there the logic behind it should be somewhere else since now it's sooo bad to have it in config but the data is not consistent etc etc.

Something to definitely refactor soon after this PR is merged.

Reply:
Yeah, we missed that the node_name was taken from the FJ which received the request...
I'll mark this with a `FIXME`

Sidenote, there are also other clear opportunities for improvement -- most notably, right now we perform the RPC twice whenever doing something with an existing room (once in `RoomService`, once in `Room`). It's totally possible to do this in a single RPC -- however, since it's the first iteration, I didn't really want to make changes this big.